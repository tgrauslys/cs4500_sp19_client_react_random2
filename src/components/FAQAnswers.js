import React from 'react'
import FAQAnswerService from '../services/FAQAnswerService'
class FAQAnswers extends React.Component {
    constructor(props) {
        super(props)
        this.faqAnswerService = FAQAnswerService.getInstance()
        this.state = {
            faqAnswers: []
        }
    }
    componentDidMount() {
        this.faqAnswerService
            .findAllFAQAnswers()
            .then(faqAnswers =>
                this.setState({
                    faqAnswers: faqAnswers
                })
            )
    }
    render() {
        return(
            <div>
                <h3>Service Answers</h3>
                <table className="table">
                    <tbody>
                    {
                        this.state.faqAnswers
                            .map(faqAnswer =>
                                <tr key={faqAnswer.id}>
                                    <td>{faqAnswer.question}</td>
                                    <td>{faqAnswer.answer}</td>
                                </tr>
                            )
                    }
                    </tbody>
                </table>
            </div>
        )
    }
}

export default FAQAnswers