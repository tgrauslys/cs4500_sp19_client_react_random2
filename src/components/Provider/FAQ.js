import React from 'react'
const FAQ = ({faq}) =>
    <div>
        <h5>{faq.question}</h5>
        <p>{faq.answer}</p>
        <hr/>
    </div>

export default FAQ