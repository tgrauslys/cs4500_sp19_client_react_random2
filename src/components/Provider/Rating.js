import React from 'react'
const Rating = ({score, index}) =>
    <div>
        <div className="row">
            <div className="col-2 text-right">
                {index}&nbsp;
                <i className="fa fa-star cs4500-yellow"/>
            </div>
            <div className="col-8">
                <div class="progress">
                    <div class="progress-bar"
                         role="progressbar"
                         style={{width: `${score}%`}}
                         aria-valuenow="25"
                         aria-valuemin="0"
                         aria-valuemax="100"></div>
                </div>
            </div>
            <div className="col-2 text-left">
                {score}%
            </div>
        </div>
    </div>

export default Rating