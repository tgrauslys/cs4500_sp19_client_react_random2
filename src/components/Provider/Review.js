import React from 'react'
const Review = ({review}) =>
    <div>
        <h4>{review.who}</h4>
        <div>
            <i className="fa fa-star cs4500-yellow"/>
            <i className="fa fa-star cs4500-yellow"/>
            <i className="fa fa-star cs4500-yellow"/>
            <i className="fa fa-star cs4500-yellow"/>
            <i className="fa fa-star cs4500-yellow"/>
        </div>
        <br/>
        <div>{review.review}</div>
        <br/>
        <div>{review.date}</div>
        <br/>
        <div class="alert alert-secondary" role="alert">
            {review.reply}
        </div>
        <hr/>
    </div>

export default Review