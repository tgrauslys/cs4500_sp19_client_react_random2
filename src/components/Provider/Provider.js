import React from 'react'
import Rating from './Rating'
import Review from './Review'
import FAQ from './FAQ'
// import Date from 'Date'


const Provider = ({provider, reviews, ratings, FAQAnswers, history}) =>
    <div>

        <div className="row">
            {/*<div className="col-8">*/}
                {/*<SearchBar/>*/}
            {/*</div>*/}
            <div className="col-3 text-right">
                <a href="#">Sign up</a>
            </div>
            <div className="col-1">
                <a href="#">Log in</a>
            </div>
        </div>
        <br/>
        <br/>
        <ul className="nav nav-pills">
            <li className="nav-item">
                <a className="nav-link" href="#about">About</a>
            </li>
            <li className="nav-item">
                <a className="nav-link" href="#reviews">Reviews</a>
            </li>
            <li className="nav-item">
                <a className="nav-link" href="#faqs">FAQs</a>
            </li>
        </ul>
        <div className="row">
            <div className="col-2">
                <a name="about"/>
                <img src="https://picsum.photos/150/150"/>
            </div>
            <div className="col-10">
                <h3>{provider.username}</h3>
                {ratings[0]}
                <i className="fa fa-star cs4500-yellow"/>
                <i className="fa fa-star cs4500-yellow"/>
                <i className="fa fa-star cs4500-yellow"/>
                <i className="fa fa-star cs4500-yellow"/>
                <i className="fa fa-star cs4500-yellow"/>
                ({reviews.length})
            </div>
        </div>
        <div>
            <br/>
            <p>
                {provider.description}
            </p>
        </div>
        <div className="row">
            <div className="col-6">
                <h4>Overview</h4>
                <i className="fa fa-trophy"/>
                &nbsp;
                Hired {provider.hires} times
                <br/>
                <i className="fa fa-lock"/>
                &nbsp;
                {provider.backgroundChecked ? 'Background checked' : ''}
                <br/>
                <i className="fa fa-users"/>
                &nbsp;
                {provider.employees} Employees
                <br/>
                <i className="fa fa-briefcase"/>
                &nbsp;
                {provider.startDate}
            </div>
            <div className="col-6">
                <h4>Payment methods</h4>
                <i className="fa fa-usd"/>
                {
                    provider.paymentMethods
                }
            </div>
        </div>
        <hr/>
        <div className="row">
            <div className="col-8">
                <a name="reviews"/>
                <h3>
                    Reviews
                </h3>
            </div>
            <hr/>
            <div className="col-4">
                <select className="form-control float-right">
                    <option>Recommended</option>
                    <option>Highest rated</option>
                    <option>Lowest rated</option>
                    <option>Newest first</option>
                    <option>Oldest first</option>
                </select>
                <br/>
                <br/>
            </div>
        </div>
        <div className="row">
            <div className="col-4">
                <h4>{ratings[0]}</h4>
                <i className="fa fa-star cs4500-yellow wd-font-size-2-em"/>
                <i className="fa fa-star cs4500-yellow wd-font-size-2-em"/>
                <i className="fa fa-star cs4500-yellow wd-font-size-2-em"/>
                <i className="fa fa-star cs4500-yellow wd-font-size-2-em"/>
                <i className="fa fa-star cs4500-yellow wd-font-size-2-em"/>
                <br/>
                {reviews.length} reviews
            </div>
            <div className="col-8">
                {
                    ratings.slice(1,5).map((score, index) =>
                        <Rating key={index}
                                index={5-index}
                                score={score}/>
                    )
                }
            </div>
        </div>
        <br/>
        <div>
            {
                reviews.map(review =>
                    <Review key={review.id} review={review}/>
                )
            }
            <div className="center">
                <nav aria-label="...">
                    <ul className="pagination">
                        <li className="page-item disabled">
                            <span className="page-link">Previous</span>
                        </li>
                        <li className="page-item"><a className="page-link" href="#">1</a></li>
                        <li className="page-item active" aria-current="page">
                              <span className="page-link">
                                2
                                <span className="sr-only">(current)</span>
                              </span>
                        </li>
                        <li className="page-item"><a className="page-link" href="#">3</a></li>
                        <li className="page-item">
                            <a className="page-link" href="#">Next</a>
                        </li>
                    </ul>
                </nav>
            </div>
        </div>
        <div>
            <hr/>
            <a name="faqs"/>
            <h3>FAQs</h3>
            {
                FAQAnswers.map(faqAnswer =>
                    <FAQ key={faqAnswer.id} faq={faqAnswer}/>
                )
            }
        </div>
    </div>

export default Provider