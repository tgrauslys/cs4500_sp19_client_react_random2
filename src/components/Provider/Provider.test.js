import React from 'react';
import Provider from './Provider';
import TestRenderer from 'react-test-renderer';
import user from '../../data/user.mock.json'
import reviews from '../../data/reviews.mock.json'
import ratingScores from '../../data/ratingScores.mock.json'
import FAQAnswers from '../../data/faqanswers.mock.json'


test('Render provider profile page', () => {
    const testRenderer = TestRenderer.create(
        <Provider
            provider={user}
            reviews={reviews}
            ratings={ratingScores}
            FAQAnswers={FAQAnswers}
            reviewDescription={""}
            reviewRating={1}

        />)

    let tree = testRenderer.toJSON()
    expect(tree).toMatchSnapshot()

    const testInstance = testRenderer.root

    const ratingWidths = testInstance.findAllByProps({class: "progress"});

    //check 100% of ratings are 5 stars
    expect(ratingWidths).toBeDefined();
    expect(ratingWidths.length).toBe(5);
    expect(ratingWidths[0].props.children.props["style"]["width"]).toBe("100%");

    //others are 0%
    expect(ratingWidths[1].props.children.props["style"]["width"]).toBe("0%");
    expect(ratingWidths[2].props.children.props["style"]["width"]).toBe("0%");
    expect(ratingWidths[3].props.children.props["style"]["width"]).toBe("0%");
    expect(ratingWidths[4].props.children.props["style"]["width"]).toBe("0%");

    

})
