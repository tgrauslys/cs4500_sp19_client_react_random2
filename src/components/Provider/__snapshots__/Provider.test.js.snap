// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render provider profile page 1`] = `
<div>
  <div
    className="row"
  >
    <div
      className="col-3 text-right"
    >
      <a
        href="#"
      >
        Sign up
      </a>
    </div>
    <div
      className="col-1"
    >
      <a
        href="#"
      >
        Log in
      </a>
    </div>
  </div>
  <br />
  <br />
  <ul
    className="nav nav-pills"
  >
    <li
      className="nav-item"
    >
      <a
        className="nav-link"
        href="#about"
      >
        About
      </a>
    </li>
    <li
      className="nav-item"
    >
      <a
        className="nav-link"
        href="#reviews"
      >
        Reviews
      </a>
    </li>
    <li
      className="nav-item"
    >
      <a
        className="nav-link"
        href="#faqs"
      >
        FAQs
      </a>
    </li>
  </ul>
  <div
    className="row"
  >
    <div
      className="col-2"
    >
      <a
        name="about"
      />
      <img
        src="https://picsum.photos/150/150"
      />
    </div>
    <div
      className="col-10"
    >
      <h3>
        luci
      </h3>
      5
      <i
        className="fa fa-star cs4500-yellow"
      />
      <i
        className="fa fa-star cs4500-yellow"
      />
      <i
        className="fa fa-star cs4500-yellow"
      />
      <i
        className="fa fa-star cs4500-yellow"
      />
      <i
        className="fa fa-star cs4500-yellow"
      />
      (
      2
      )
    </div>
  </div>
  <div>
    <br />
    <p />
  </div>
  <div
    className="row"
  >
    <div
      className="col-6"
    >
      <h4>
        Overview
      </h4>
      <i
        className="fa fa-trophy"
      />
        Hired 
       times
      <br />
      <i
        className="fa fa-lock"
      />
       
      
      <br />
      <i
        className="fa fa-users"
      />
       
       Employees
      <br />
      <i
        className="fa fa-briefcase"
      />
       
    </div>
    <div
      className="col-6"
    >
      <h4>
        Payment methods
      </h4>
      <i
        className="fa fa-usd"
      />
    </div>
  </div>
  <hr />
  <div
    className="row"
  >
    <div
      className="col-8"
    >
      <a
        name="reviews"
      />
      <h3>
        Reviews
      </h3>
    </div>
    <hr />
    <div
      className="col-4"
    >
      <select
        className="form-control float-right"
      >
        <option>
          Recommended
        </option>
        <option>
          Highest rated
        </option>
        <option>
          Lowest rated
        </option>
        <option>
          Newest first
        </option>
        <option>
          Oldest first
        </option>
      </select>
      <br />
      <br />
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-4"
    >
      <h4>
        Rating: 5
      </h4>
      <i
        className="fa fa-star cs4500-yellow wd-font-size-2-em"
      />
      <i
        className="fa fa-star cs4500-yellow wd-font-size-2-em"
      />
      <i
        className="fa fa-star cs4500-yellow wd-font-size-2-em"
      />
      <i
        className="fa fa-star cs4500-yellow wd-font-size-2-em"
      />
      <i
        className="fa fa-star cs4500-yellow wd-font-size-2-em"
      />
      <br />
      2
       reviews
    </div>
    <div
      className="col-8"
    >
      <div>
        <div
          className="row"
        >
          <div
            className="col-2 text-right"
          >
            5
             
            <i
              className="fa fa-star cs4500-yellow"
            />
          </div>
          <div
            className="col-8"
          >
            <div
              class="progress"
            >
              <div
                aria-valuemax="100"
                aria-valuemin="0"
                aria-valuenow="25"
                class="progress-bar"
                role="progressbar"
                style={
                  Object {
                    "width": "100%",
                  }
                }
              />
            </div>
          </div>
          <div
            className="col-2 text-left"
          >
            100
            %
          </div>
        </div>
      </div>
      <div>
        <div
          className="row"
        >
          <div
            className="col-2 text-right"
          >
            4
             
            <i
              className="fa fa-star cs4500-yellow"
            />
          </div>
          <div
            className="col-8"
          >
            <div
              class="progress"
            >
              <div
                aria-valuemax="100"
                aria-valuemin="0"
                aria-valuenow="25"
                class="progress-bar"
                role="progressbar"
                style={
                  Object {
                    "width": "0%",
                  }
                }
              />
            </div>
          </div>
          <div
            className="col-2 text-left"
          >
            0
            %
          </div>
        </div>
      </div>
      <div>
        <div
          className="row"
        >
          <div
            className="col-2 text-right"
          >
            3
             
            <i
              className="fa fa-star cs4500-yellow"
            />
          </div>
          <div
            className="col-8"
          >
            <div
              class="progress"
            >
              <div
                aria-valuemax="100"
                aria-valuemin="0"
                aria-valuenow="25"
                class="progress-bar"
                role="progressbar"
                style={
                  Object {
                    "width": "0%",
                  }
                }
              />
            </div>
          </div>
          <div
            className="col-2 text-left"
          >
            0
            %
          </div>
        </div>
      </div>
      <div>
        <div
          className="row"
        >
          <div
            className="col-2 text-right"
          >
            2
             
            <i
              className="fa fa-star cs4500-yellow"
            />
          </div>
          <div
            className="col-8"
          >
            <div
              class="progress"
            >
              <div
                aria-valuemax="100"
                aria-valuemin="0"
                aria-valuenow="25"
                class="progress-bar"
                role="progressbar"
                style={
                  Object {
                    "width": "0%",
                  }
                }
              />
            </div>
          </div>
          <div
            className="col-2 text-left"
          >
            0
            %
          </div>
        </div>
      </div>
      <div>
        <div
          className="row"
        >
          <div
            className="col-2 text-right"
          >
            1
             
            <i
              className="fa fa-star cs4500-yellow"
            />
          </div>
          <div
            className="col-8"
          >
            <div
              class="progress"
            >
              <div
                aria-valuemax="100"
                aria-valuemin="0"
                aria-valuenow="25"
                class="progress-bar"
                role="progressbar"
                style={
                  Object {
                    "width": "0%",
                  }
                }
              />
            </div>
          </div>
          <div
            className="col-2 text-left"
          >
            0
            %
          </div>
        </div>
      </div>
    </div>
  </div>
  <br />
  <div>
    <div>
      <h4>
        Kristine M.
      </h4>
      <div>
        <i
          className="fa fa-star cs4500-yellow"
        />
        <i
          className="fa fa-star cs4500-yellow"
        />
        <i
          className="fa fa-star cs4500-yellow"
        />
        <i
          className="fa fa-star cs4500-yellow"
        />
        <i
          className="fa fa-star cs4500-yellow"
        />
      </div>
      <div>
         
         
      </div>
      <br />
      <br />
      <div />
      <br />
      <div>
        Apr 13, 2017
      </div>
      <br />
      <hr />
    </div>
    <div>
      <h4>
        Eric G.
      </h4>
      <div>
        <i
          className="fa fa-star cs4500-yellow"
        />
        <i
          className="fa fa-star cs4500-yellow"
        />
        <i
          className="fa fa-star cs4500-yellow"
        />
        <i
          className="fa fa-star cs4500-yellow"
        />
        <i
          className="fa fa-star cs4500-yellow"
        />
      </div>
      <div>
         
         
      </div>
      <br />
      <br />
      <div />
      <br />
      <div>
        Apr 19, 2017
      </div>
      <br />
      <hr />
    </div>
    <div
      className="center"
    >
      <nav
        aria-label="..."
      >
        <ul
          className="pagination"
        >
          <li
            className="page-item disabled"
          >
            <span
              className="page-link"
            >
              Previous
            </span>
          </li>
          <li
            className="page-item"
          >
            <a
              className="page-link"
              href="#"
            >
              1
            </a>
          </li>
          <li
            aria-current="page"
            className="page-item active"
          >
            <span
              className="page-link"
            >
              2
              <span
                className="sr-only"
              >
                (current)
              </span>
            </span>
          </li>
          <li
            className="page-item"
          >
            <a
              className="page-link"
              href="#"
            >
              3
            </a>
          </li>
          <li
            className="page-item"
          >
            <a
              className="page-link"
              href="#"
            >
              Next
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </div>
  <div>
    <input
      className="review-fld"
      name="reviewDescription"
      value=""
    />
    <select
      className="star-fld"
      name="reviewRating"
      value={1}
    >
      <option
        value={1}
      >
        1
      </option>
      <option
        value={2}
      >
        2
      </option>
      <option
        value={3}
      >
        3
      </option>
      <option
        value={4}
      >
        4
      </option>
      <option
        value={5}
      >
        5
      </option>
    </select>
    <button
      className="submit-review-btn"
    >
      Submit Review
    </button>
  </div>
  <div>
    <hr />
    <a
      name="faqs"
    />
    <h3>
      FAQs
    </h3>
    <div>
      <h5>
        What should the customer know about your pricing (e.g., discounts, fees)?
      </h5>
      <p>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum
      </p>
      <hr />
    </div>
    <div>
      <h5>
        What education and/or training do you have that relates to your work?
      </h5>
      <p>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum
      </p>
      <hr />
    </div>
  </div>
</div>
`;
